frontend/components/exercise/AudioPlayer.tsx

```diff
@@ -44,66 +39,87 @@ export default function AudioPlayer({ audioSrc, title }) {
-    : calculatedDashOffset;
-  lastValidDashOffset.current = dashOffset;
+  const radius = 64;
+  const circumference = 2 * Math.PI * radius;
+  const progress = (currentTime / duration) * circumference;
 
   const formatTime = (time) => {
     const minutes = Math.floor(time / 60);
     const seconds = Math.floor(time % 60);
     return `${minutes}:${seconds.toString().padStart(2, "0")}`;
   };
 
-  // Single return statement with conditional rendering
-  return isMobile ? (
-    <AudioPlayerMobile audioSrc={audioSrc} title={title} />
-  ) : (
-    <div className="relative h-[128px] w-[128px]">
-      <div className="relative flex h-[128px] w-[128px] items-center justify-center">
-        <svg
-          className="absolute center-atop-svg"
-          width="128"
-          height="128"
-          viewBox="0 0 128 128"
-          fill="none"
-          xmlns="http://www.w3.org/2000/svg"
-        >
-          <circle
-            cx={64}
-            cy={64}
-            r={radius}
-            strokeWidth="2px"
-            stroke="var(--surface_bg_secondary)"
-            fill="var(--surface_bg_highlight)"
-          />
-          <circle
-            cx={64}
-            cy={64}
-            r={radius}
-            strokeWidth="2px"
-            stroke="var(--surface_bg_dark)"
-            style={{
-              strokeDasharray: dashArray,
-              strokeDashoffset: dashOffset,
-              transition: "stroke-dashoffset 0.05s linear",
-            }}
-            transform={`rotate(-90 64 64)`}
-          />
-        </svg>
-      </div>
-      <div className="absolute inset-0 flex items-center justify-center">
-        <AudioButton size={40} togglePlay={togglePlay} isPlaying={isPlaying} />
-      </div>
-      <div className="absolute bottom-[22px] left-0 flex w-full items-center justify-center">
+  return (
+    <div className="relative h-32 w-32">
+      {/* Outer progress bar */}
+      <svg
+        className="absolute inset-0"
+        width="128"
+        height="128"
+        viewBox="0 0 128 128"
+      >
+        <circle
+          cx="64"
+          cy="64"
+          r={radius}
+          className="fill-none stroke-surface_bg_secondary"
+          strokeWidth="4"
+        />
+        <circle
+          cx="64"
+          cy="64"
+          r={radius}
+          className="fill-none stroke-surface_bg_darker"
+          strokeWidth="4"
+          strokeDasharray={circumference}
+          strokeDashoffset={circumference - progress}
+          transform="rotate(-90 64 64)"
+          style={{ transition: "stroke-dashoffset 0.05s linear" }}
+        />
+      </svg>
+
+      {/* Inner circle */}
+      <div className="absolute inset-0.5 rounded-full bg-white"></div>
+
+      {/* Button with SVG */}
+      <button
+        onClick={togglePlay}
+        className="absolute inset-0 m-auto flex h-11 w-11 items-center justify-center rounded-full bg-action_bg_primary text-action_bg_secondary shadow-[0_4px_20px_rgba(0,0,0,0.05)]"
```

Comment:
use the custom tailwind shadows please. much easier and consistent. also, use arbitrary values h-[44px] and w-[44px] to be consistent; don't wanna be multiplying by 4 to figure out if things are the right size